src/functions-reference/higher-order_functions.Rmd

```diff
@@ -141,6 +142,138 @@ Stan's automatic differentiation.
 
 ## Ordinary Differential Equation (ODE) Solvers {#functions-ode-solver}
 
+Stan provides built-in ordinary differential equation (ODE) solvers.
+Because the first argument to each of the solvers is a function itself,
+the ODE solvers in Stan are considered high order functions.
+
+### Specifying an Ordinary Differential Equation as a Function
+
+A system of ODEs is specified as an ordinary function in Stan within
+the functions block. The ODE system function must have the same return
+type and first two arguments as the following function:
+
+```
+ vector ode(real time, vector state, ...)
+```
+
+`...` are any number of arguments passed through unmodified from the ODE integrate
+call.
+
+The ODE system function should return the derivative of the state with
+respect to time at the time and state provided. The length of the returned
+vector must match the length of the state input into the function.
+
+The arguments to this function are:
+
+*   *`time`*, the time to evaluate the ODE system
+
+*   *`state`*, the state of the ODE system at the time specified
+
+*   *...*, all other arguments passed through from the integrate call. Each
+argument can be any type.
+
+### Non-Stiff Solver
+
+<!-- vector[]; ode_rk45; (function ode, vector initial_state, real initial_time, real[] times, ...); -->
+\index{{\tt \bfseries ode\_rk45 }!{\tt (function ode, real[] initial\_state, real initial\_time, real[] times, ...): vector[]}|hyperpage}
+
+`vector[]` **`ode_rk45`**`(function ode, vector initial_state, real initial_time, real[] times, ...)`<br>\newline
+Solves the ODE system for the times provided using the Runge Kutta
+Dopri algorithm with the implementation from Boost.
```

Comment:
Should there be a link to wiki or boost for this method? Or is the user expected to know this solver already?

Reply:
I made the naming a little clearer, but yeah this is a searchable standard algorithm.